mobile/src/screens/marketplace.js

```diff
@@ -108,16 +101,56 @@ class MarketplaceScreen extends Component {
       return
     }
 
-    if (this[msgData.targetFunc]) {
+    const currentRoute = getCurrentRoute()
+
+    if (msgData.targetFunc === 'getAccounts') {
+      // Call get account method from OriginWallet HOC
+      const response = this.props.getAccounts()
+      this.handleBridgeResponse(msgData, response)
+    } else if (this[msgData.targetFunc]) {
       // Function handler exists, use that
       const response = this[msgData.targetFunc].apply(this, [msgData.data])
       this.handleBridgeResponse(msgData, response)
+    } else if (msgData.targetFunc === 'signPersonalMessage') {
+      // Personal sign is for handling meta transaction requests
+      const decodedData = JSON.parse(
+        global.web3.utils.hexToUtf8(msgData.data.data)
+      )
+      const decodedTransaction = decodeTransaction(decodedData.txData)
+      // If the transaction validate the sha3 hash and sign that for the relayer
+      if (this.isValidMetaTransaction(decodedTransaction)) {
+        const dataToSign = global.web3.utils.soliditySha3(
```

Comment:
Again seems like this kind of logic should happen in either an action or a reducer but no big deal